docs/vagrant.md

```diff
@@ -0,0 +1,134 @@
+
+## Notes
+
+This was tested using:
+
+- Vagrant v2.4.0
+- VirutalBox v7.0.12
+
+## Prerequisites
+
+1. You will require 4 GB of memory.
+1. You will require 8 GB of disk space.
+1. Ports 4200, 4300, and 5432 must be available on host environment. If you have powershell installed you can use this command to list active ports:
+
+   ```powershell
+   Get-NetTCPConnection | Where-Object { $_.State -eq 'Listen' }
+   ```
+
+1. `npm install` creates symbolic links. As a result, the VM user: `vagrant` requires `create symbolic link` authority. In some environements, only the administrator has this authority. To resolve this issue in Windows, the following steps can be performed:
+   1. Launch the `Local Security Policy` app from the `start menu` (or `Win+R`, then type `secpol.msc`)
+   1. Navigate to `Local Policies` -> `User Rights Assignment`.
+   1. Open the `Create symbolic links` property. By default it has only `Administrators` listed.
+   1. Click `Add User or Group...` button and type `Authenticated Users` in the `Enter the object names to select` field and then press the `OK` button. This will grant `symbolic link` priviledge to all users who log on with credetials.
+   1. Close any open windows by clicking `OKs` and and then `log off` and `log on` to activate.
+   1. Symbolic links should now work in VirtualBox shared folders without administrator priviledges.
+
+## Setup Steps
+
+1. [Download](https://developer.hashicorp.com/vagrant/downloads?ajs_aid=ba208f95-d5a1-457d-abeb-49d458b95eec&product_intent=vagrant) Vagrant software.
+1. Reference these [installation](https://developer.hashicorp.com/vagrant/docs/installation) instructions.
+1. Verify installation by running `vagrant -v` in a terminal session.
+1. Install [VirtualBox](https://www.virtualbox.org/).
+1. If you have not already done so, fork repo to your local machine.
+1. Vagrant uses a special configuration file called: `Vagrantfile` that contains all the information required for the creation of a Virtual Machine (guest) on your workstation (host) . It is located n your repo root directory.
+1. To start the creation of your virtual environment, run this command in the repo root directory. (This takes a while so be patient.)
+
+   ```bash
+   vagrant up
+   ```
+
+1. Once the `vagrant up` command completes, you can logon into your virutal machine using this command. (password: `vagrant`)
+
+   ```bash
+   vagrant ssh
+   ```
+
+1. Here is a list of useful Vagrant commands. Be sure that you are in the same directory as the Vagran­tfile when running these commands!
```

Comment:
Is it necessary to have this list, when we can just use `vagrant --help` to get the available commands? Having this list in our docs requires us to keep it in sync with vagrant's docs. If they update or remove a command, our docs will be out of date.

Reply:
Some of the items like `VAGRANT_LOG=info` would be useful for our docs, because they are not listed in `vagrant --help`

Reply:
Maybe we should include the very basic commands like `vagrant up`, `vagrant halt`, and `vagrant ssh` (basically only the most basic commands which we'll need for day-to-day usage). Then we can refer developers to `vagrant --help` if they want to go deeper.

@arrocke What are your thoughts? Am I way off?

Reply:
Yep, this documentation should be enough for a developer to get started, and refer to the vagrant docs for more detail. I think we can assume that devs that want to use vagrant, will do their own research or will already be familiar

Reply:
So just to be clear. Do you want me to remove point 9.  Here is a list of usefule Vagrant commands....    but leave point 10?

Reply:
I'm ok removing both. Devs can do their own research if they want to do more than the basic commands to get everything running

Reply:
@ilearner777 Go ahead and remove points 9 and 10. You should replace them with a point that describes the basic commands, just "enough for a developer to get started" like Adrian said.

Reply:
done